### PR TITLE
Add frontpress@1.0.8 w/ npm auto-update

### PR DESCRIPTION
**Pull request by teles**
Pull request for issue: #10236
Related issue(s): #10187 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/frontpressorg/frontpress
 * Official website (optional, not the repository): https://github.com/frontpressorg/frontpress
 * NPM package url (optional): https://www.npmjs.com/package/frontpress
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 68
   - Count of watchers: 10
   - Count of forks: 11
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month: 94

# Essential checklist
 * [x] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by teles** [comment]
@PeterDaveHello I don't know how to squase commits into one from github interface.

**Comment by PeterDaveHello** [comment]
@teles no worry, I can squash that for you, there is no way that you can do it via GitHub gui right now.

**Comment by PeterDaveHello** [comment]
@teles you need to move the package.json outside the version directory.

**Comment by x09326** [comment]
@teles You can use the **fixformat.js** in the **cdnjs/tools** folder to remove some fields that we don't need in **package.json**

File: ajax/libs/frontpress/1.0.7/package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    "type": "git",
+    "url": "git+https://github.com/frontpressorg/frontpress.git"
+  },  
+  "author": "Jota Teles",
+  "npmName": "frontpress",
+  "npmFileMap": [
+    {
+      "basePath": "release",
+      "files": [
+        "**/frontpress.*.js"
```
Comment (pvnr0082t) [inline ajax/libs/frontpress/1.0.7/package.json]: @teles We can use `"**/*"` to grab all files under release folder.

**Comment by teles** [comment]
@PeterDaveHello @kennynaoh @x09326 i did the requested changes in package.json file.

**Comment by PeterDaveHello** [comment]
hmmmm ... interesting, the test result failed and said - `Could not find any tests to run.`

@cdnjs/intern2 guys can you try to run this test locally and find out the problem?

**Comment by PeterDaveHello** [comment]
oops, I should find out the problem, will restart the test.

**Comment by PeterDaveHello** [comment]
CI issue solved, let's fix its indent to 2-space indent!

**Comment by teles** [comment]
indentation of package.json file fixed for 2 spaces, @PeterDaveHello :)

File: ajax/libs/frontpress/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "name": "Jota Teles",
+    "url": "https://github.com/teles"
+  },
+  "npmName": "frontpress",
+  "npmFileMap": [{
+    "basePath": "release",
+    "files": [
+      "**/*"
+    ]
+  }],
```
Comment (PeterDaveHello) [inline ajax/libs/frontpress/package.json]: This part is still wrong, would you please use tool - `./tools/fixFormat.js` to fix it for you?

Comment (PeterDaveHello) [inline ajax/libs/frontpress/package.json]: Or you are using GitHub gui?

File: ajax/libs/frontpress/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/frontpressorg/frontpress.git"
+  },
+  "author": {
+    "name": "Jota Teles",
+    "url": "https://github.com/teles"
+  },
+  "npmName": "frontpress",
+  "npmFileMap": [{
```
Comment (kennynaoh) [inline ajax/libs/frontpress/package.json]: Hello @teles  `[{` wrong format here. Could you run the fix tools `fixFormat.js` first https://github.com/cdnjs/cdnjs/tree/master/tools. Thanks!

**Comment by teles** [comment]
Done. I ran fixFormat for package.json file.

File: ajax/libs/frontpress/package.json (hunk 2 above)
Comment (teles) [inline ajax/libs/frontpress/package.json]: Hi, @PeterDaveHello . Now I ran fixFormat.

File: ajax/libs/frontpress/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "url": "git+https://github.com/frontpressorg/frontpress.git"
+  },
+  "author": {
+    "name": "Jota Teles",
+    "url": "https://github.com/teles"
+  },
+  "npmName": "frontpress",
+  "npmFileMap": [
+    {
+      "basePath": "release",
```
Comment (PeterDaveHello) [inline ajax/libs/frontpress/package.json]: How come I didn't see `release` folder there? @x09326 @kennynaoh @pvnr0082t ?

**Comment by teles** [comment]
@PeterDaveHello  do I need to include all js files inside a directory named ` ajax/libs/frontpress/1.0.7/release` in this PR?

**Comment by PeterDaveHello** [comment]
@teles nope, the base path means it'll be treated as the root of a version to grab the files, the problem is, is there a release directory under its npm package? I didn't see it.

**Comment by teles** [comment]
I tested here now doing 
`npm install frontpress`;
 and then as you verified there is no `release` folder :( 
I guess this is happening because `release` folder is in [.gitignore](https://github.com/frontpressorg/frontpress/blob/master/.gitignore)

I will remove release from gitignore and update npm version to 1.0.8 and update this PR

**Comment by teles** [comment]
@PeterDaveHello now there is a release file.
Tested here doing `npm install frontpress` 
NPM package updated. Thanks for pointing this problem!

**Comment by PeterDaveHello** [comment]
@cdnjs/intern2 so all your reviews have the problem, right? Can you explain about what's going on that all of you three didn't confirm the config with upstream?

**Comment by pvnr0082t** [comment]
@PeterDaveHello I use unpkg to see the files but after I change to old versions, the latest version(v1.0.7) disappeared and I didn't notice that. So what I saw the structure was v1.0.6, it's a mistake.

**Comment by kennynaoh** [comment]
@PeterDaveHello Sorry for my mistake.

**Comment by PeterDaveHello** [comment]
@kennynaoh So what's going on?

**Comment by teles** [comment]
@PeterDaveHello how can I help this PR to be approved ?

**Comment by PeterDaveHello** [comment]
@teles We'll make sure everything is okay and merge it soon, hope there is nothing else need to be modified, thanks!

**Comment by teles** [comment]
Thank you, @PeterDaveHello

**Comment by kennynaoh** [comment]
@PeterDaveHello I have some mistakes on my review. I will be more careful next time.

**Comment by PeterDaveHello** [comment]
@x09326 same situation?

**Comment by x09326** [comment]
@PeterDaveHello Sorry that I didn't download the branch, I just use unpkg to see the files, so I didn't notice the release folder can't be found

**Comment by PeterDaveHello** [comment]
@x09326 Nope, that's not about you download it or not ... can't you see the folder online?

**Comment by PeterDaveHello** [comment]
@x09326 So how did you confirm the auto-update config in the past?

**Comment by x09326** [comment]
@PeterDaveHello If the versions are fewer, I just use unpkg to check, but I found that  I missed the v1.0.7 there is no `release` folder. I will fetch the branch to confirm the auto-update config in the future.

**Comment by PeterDaveHello** [comment]
@x09326 I think you don't need to 'fetch' it since the problem you got this time can actually easily be catched by eyes, the point may be just being much more careful and seriously next time.

**Comment by PeterDaveHello** [comment]
@cdnjs/intern2 let's make a serious review and get this PR merged asap, are you good with the latest status now?

**Comment by x09326** [comment]
LGTM
Just notice that the npm can't get version v1.0.2~v.1.0.4 and v1.0.7

**Comment by PeterDaveHello** [comment]
@teles I have a question, looks like `frontpress.js` is the same as `frontpress.v1.js`, same situation on the minified ones, do we need to host the same file twice?

**Comment by teles** [comment]
Hi, @PeterDaveHello.  `frontpress.js` and `frontpress.v1.js` are not the same file.
[DiffChecker found the difference between files](https://www.diffchecker.com/Jm4jc3EK)
[DiffChecker for minified files](https://www.diffchecker.com/O9oIKYM8) 

Example comparision for function `getPostBySlug`: 

[frontpress.js](https://raw.githubusercontent.com/frontpressorg/frontpress/master/release/frontpress.js) is a release file for WordPress API v2.

```javascript
    function getPostBySlug(postSlug, configs){
        var postUrl = postsBaseUrl + "?filter[name]=\"<post-slug>\"";
        postUrl = postUrl.replace("<post-slug>", postSlug);

        return AjaxModel.get(postUrl, configs);
    }
```

[frontpress.v1.js](https://raw.githubusercontent.com/frontpressorg/frontpress/master/release/frontpress.v1.js) is a release file for WordPress API v1 (deprecated but still in use).

```javascript
    function getPostBySlug(postSlug, configs){
        var postUrl = postsBaseUrl + "slug:<post-slug>";
        postUrl = postUrl.replace("<post-slug>", postSlug);
        return AjaxModel.get(postUrl, configs);
    }
``` 

[This test commit shows difference between files](https://github.com/frontpressorg/frontpress/commit/6c0782032b5d5706ec36ff55adbf28919b227fbc#diff-d2595d1510af8a32fbbae52d753daa9d), in this commit I replaced `frontpress.js` content with `frontpress.v1.js` (it's just a test outside master branch).


So, it would be nice to have the four files hosted, but if I need to choose just two of them I would choose to host `frontpress.js` and `frontpress.min.js`.
Thanks.

**Comment by PeterDaveHello** [comment]
@teles Oops, my fault, I must remember the wrong lib, thanks.

**Comment by teles** [comment]
Thanks @PeterDaveHello . FrontPress is merged now 🎉  🎉

**Comment by PeterDaveHello** [comment]
@teles Thank you, too. It'll be availible on https://cdnjs.com/libraries/frontpress in 15 mins!

**Comment by PeterDaveHello** [comment]
@teles before that, it's already online on the CDN infra:
 - https://cdnjs.cloudflare.com/ajax/libs/frontpress/1.0.8/frontpress.js
 - https://cdnjs.cloudflare.com/ajax/libs/frontpress/1.0.8/frontpress.min.js

**Comment by PeterDaveHello** [comment]
@teles It's also online on the website now!

**Comment by teles** [comment]
So nice, thank you @PeterDaveHello and all the team.

**Comment by PeterDaveHello** [comment]
@teles Thank you, too :)

**Pull request merged**